### PR TITLE
RIP 7728: L1SLOAD precompile

### PR DESCRIPTION
**Pull request by icemelon**
This proposal introduces a new precompiled contract `L1SLOAD` to L2s that can load storage slots from L1 given a contract address and storage keys.

### CODE REVIEW

File: RIPS/rip-tbd.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+---
+rip: XXXX
```
Comment (abcoathup) [inline RIPS/rip-tbd.md:2]:
```suggestion
rip: 7728
```
Assigning next sequential EIP/ERC/RIP number.

Please also update the filename.

File: RIPS/rip-tbd.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+title: Precompile for L1SLOAD
+description: Proposal to add precompiled contract that load L1 storage slots.
+author: Haichen Shen (@icemelon), Péter Garamvölgyi (@Thegaram)
+discussions-to:
```
Comment (abcoathup) [inline RIPS/rip-tbd.md:6]: Please create a discussions topic in Eth Magicians with a link back to this PR

File: RIPS/rip-tbd.md (hunk 1 above)
Comment (icemelon) [reply RIPS/rip-tbd.md]: Thank you. Updated.

File: RIPS/rip-tbd.md (hunk 2 above)
Comment (icemelon) [reply RIPS/rip-tbd.md]: created: https://ethereum-magicians.org/t/rip-7728-l1sload-precompile/20388

File: RIPS/rip-7728.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+### Precompile
+
+The inputs to the `L1SLOAD` precompile are an L1 contract address and $k$ storage keys up to `MAX_NUM_STORAGE_SLOTS`.
+
+| Byte range | Name | Description |
+| --- | --- | --- |
+| [0: 19] (20 bytes) | `address` | The contract address |
+| [20: 51] (32 bytes) | `key1` | The storage key |
+| ... | ... | ... |
+| [`k`\*32-16: `k`\*32+16] (32 bytes) | `keyk` | The storage key |
```
Comment (NazariiDenha) [inline RIPS/rip-7728.md:46]:
```suggestion
| [`k`\*32-12: `k`\*32+19] (32 bytes) | `keyk` | The storage key |
```

**Comment by gakonst** [comment]
Thanks Haichen! Is the archive node RPC requirement reasonable?

**Comment by icemelon** [comment]
@gakonst The L1SLOAD precompile does introduce additional requirements to the nodes. It is only required when a new L2 follower node is set up. After the node has caught up to the latest block, it no longer needs an L1 archive node anymore since `L1SLOAD` precompile only returns the storage value from the latest L1 blocks known to the L2 chain, which should be within the 256 blocks range.

**Review by CarlBeek — APPROVED** [review]

**Pull request merged**